### PR TITLE
Sonarlint updates to ConfirmationCodeModelMapper class.

### PR DESCRIPTION
**Pull request by CanadianPhilipBarkow**
### Description

Updates made to `ConfirmationCodeModelMapper` class including:

- Null pointer check added.
- Lambda replaced with method reference.

Relevant test updates made.

### Related Azure Boards Work Items

[AB#4173](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4173)
[AB#4178](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4178)

### Checklist

- [X] I have tested the changes locally
- [X] I have added/updated tests that prove my fix is effective or that my feature works
- [X] I have checked that my code follows the project's coding style by running `npm run format:check`
- [X] I have checked that my code contains no linting errors by running `npm run lint`
- [X] I have checked that my code contains no type errors by running `npm run typecheck`
- [X] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [X] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java [hunk 1]
```diff
@@ -17,11 +17,14 @@ public abstract class ConfirmationCodeModelMapper extends AbstractModelMapper {
...
+		if (confirmationCodes == null) {
+			return CollectionModel.empty();
+		}
 		final var confirmationCodeModels = StreamSupport.stream(confirmationCodes.spliterator(), false)
-			.map(confirmationCode -> toModel(confirmationCode)).toList();
+			.map(this::toModel).toList();
 
 		return wrapCollection(CollectionModel.of(confirmationCodeModels), ConfirmationCodeModel.class);
-	}	
+	}
```
Comment (gregory-j-baker) [inline backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java:27]: I actually think this entire method should be removed, along with the controller method that calls it.

Exposing a method to allow the consumer to enumerate the confirmation codes for a user creates an exploitable loophole where a user's email can be validated without ever using OOBA.

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		assertThat(confirmationCodeModelMapper.toModel("d827416b-f808-4035-9ccc-7572f3297015", null)).isEmpty();
+	}
+	@Test
+	final void testToModelStringIterableOfConfirmationCode_NonNullConfirmationCode() {
+		final var confirmationCodes = new HashSet<ConfirmationCode>() {{
+			add(ImmutableConfirmationCode.builder().id("00000000-0000-0000-0000-000000000000").build());
+			add(ImmutableConfirmationCode.builder().id("22222222-2222-2222-2222-222222222222").build());
+		}};
+		assertThat(confirmationCodeModelMapper.toModel("d827416b-f808-4035-9ccc-7572f3297015", confirmationCodes)).isNotEmpty();
+	}
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapperTest.java:31]: Create maps using `Map.of(k1, v1, k2, v2, etc)`

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java (hunk 1 above)
Comment (gregory-j-baker) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java:27]: I created a workitem for this: https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4235

Comment (nicholas-ly) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java:27]: The controller method is exposed to add a `link` element to user's `confirmation-codes` resource:
https://github.com/DTS-STN/canadian-dental-care-plan/blob/c8478738b41d4b0330952b25e94ec3c483328e5a/backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/UserModelMapper.java#L48

I feel the controller method should return something like a 405 (Method Not Allowed) instead.

Comment (gregory-j-baker) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java:27]: Yes.. good point @nicholas-ly

Comment (CanadianPhilipBarkow) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java:27]: > I created a workitem for this: https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4235

Okay, in that case I'll just close this PR and start on the next task.

**Pull request closed without merging**